controllers/tests/cruisecontroloperation_ttl_controller_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright © 2022 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/banzaicloud/koperator/api/v1alpha1"
+	"github.com/banzaicloud/koperator/api/v1beta1"
+	"github.com/banzaicloud/koperator/controllers"
+	"github.com/banzaicloud/koperator/pkg/util"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+var _ = Describe("CruiseControlTaskReconciler", func() {
+	var (
+		count              uint64 = 0
+		namespace          string
+		namespaceObj       *corev1.Namespace
+		kafkaClusterCRName string
+		kafkaCluster       *v1beta1.KafkaCluster
+		opName             string = "operation1"
+	)
+	BeforeEach(func() {
+		atomic.AddUint64(&count, 1)
+
+		namespace = fmt.Sprintf("cruisecontroloperationttl-%v", count)
+		namespaceObj = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+		kafkaClusterCRName = controllers.CruiseControlTaskTestKafkaClusterName
+		kafkaCluster = createMinimalKafkaClusterCR(kafkaClusterCRName, namespace)
+	})
+
+	JustBeforeEach(func() {
+		By("creating namespace " + namespace)
+		err := k8sClient.Create(context.Background(), namespaceObj)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating kafka cluster object " + kafkaCluster.Name + " in namespace " + namespace)
+		err = k8sClient.Create(context.Background(), kafkaCluster)
+		Expect(err).NotTo(HaveOccurred())
+
+	})
+
+	When("there is finished (completed) operation with TTL", func() {
+		JustBeforeEach(func() {
+			operation := generateCruiseControlOperation(opName, namespace, kafkaCluster.GetName())
+			operation.Spec.TTLSecondsAfterFinished = util.IntPointer(5)
+			err := k8sClient.Create(context.Background(), &operation)
+			Expect(err).NotTo(HaveOccurred())
+
+			operation.Status.CurrentTask = &v1alpha1.CruiseControlTask{
+				ID:        "12345",
+				Operation: v1alpha1.OperationAddBroker,
+				State:     v1beta1.CruiseControlTaskCompleted,
+				Finished:  &metav1.Time{Time: time.Now().Add(-time.Second*v1alpha1.DefaultRetryBackOffDurationSec - 10)},
+			}
+			err = k8sClient.Status().Update(context.Background(), &operation)
+			Expect(err).NotTo(HaveOccurred())
+
+		})
+		FIt("it should  remove the finished CruiseControlOperation", func() {
```

Comment:
```suggestion
		FIt("it should remove the finished CruiseControlOperation", func() {
```
**Question**: why this is `FIt()`, shouldn't it be just `It()`?